frontend/src/components/ui/SelectedTowerInfo.jsx

```diff
@@ -0,0 +1,36 @@
+import { TERRAFORMS } from "../../entities/buildables";
+
+export const SelectedTowerInfo = ({selectedTower, purchasingItem, setSelectedTower}) => {
+    if (purchasingItem?.name.startsWith("terraform")) {
+        const terraform = TERRAFORMS.get(purchasingItem.name);
+        return <div>
+            <h5>Selected Item</h5>
+            <p>Type: {terraform.label} Terrain</p>
+            <p>Price: {terraform.price}</p>
+            {
+                terraform.label === "Fill"
+                ? <p>Increase the height of a terrain tile, forcing enemies to climb over or path around.</p>
+                : <p>Remove a terrain tile (until bedrock), forcing enemies to climb down or path around.</p>
+            }
+        </div>
+    }
+
+    if (!selectedTower) {
+        return <div>
+            <h5>Selected Tower</h5>
+            <p>No tower selected</p>
+        </div>
+    }
+
+    const { name, price, damage, cooldown, minRange, maxRange } = selectedTower
+    return <div onClick={() => setSelectedTower(null)}>
```

Comment:
I wanted that too, but had to sleep on it to think of the implementation. I think I got it, gotta have the selectTower onClick Callback setup another onClick that deselects.

Reply:
JK, it was not that simple. I tried the above and adding `if (selectedTower) setSelectedTower(null)` to the click callback, but neither worked easily.

Reply:
Huh, can't you just make it toggle if the tower clicked is already selected?
`if (tower === selectedTower) deselect...`

Reply:
I haven't been able to get it yet. It seems like the callback function doesn't have access to `selectedTower`. Could it be something tricky with `mouseInput`? No other ideas atm.
![image](https://github.com/user-attachments/assets/8411ecc2-25a8-4aa9-a626-b0e38e094f5a)
![image](https://github.com/user-attachments/assets/35337a77-bbec-4f53-bf9a-6801363602ed)

Reply:
Ah, I guess we would need a level of indirection or to store it somewhere else. The value of `selectedTower` at the time the callback is created is probably `null`, so that would make sense.